### DISCUSSION TITLE
Eddy probe support

### DISCUSSION THREAD

**Discussion by l3ean5** [jschuh/klipper-macros#246] [Q&A] [answered]
I was wondering if there's something I can change so that I can get the most out of my eddy sensor.  If I put the right prompt in the console (BED_MESH_CALIBRATE METHOD=scan SCAN_MODE=rapid) it'll build a mesh in like 5 seconds, but using these macros it does a bltouch style probe before every print.  Any help would be appreciated.  Thanks.

**Comment by jschuh** [comment]
You can append the arguments with a simple macro wrapper. Something like this:

```
[gcode_macro BED_MESH_CALIBRATE]
rename_existing: _BED_MESH_CALIBRATE
gcode:
  _BED_MESH_CALIBRATE METHOD=scan SCAN_MODE=rapid {rawparams}

```

**Reply by l3ean5** [reply]
Hate to revive this, but can you explain to me how to get this to work or where I have to add this?  I just installed the eddy on another printer but I can't remember what I did before.  At the moment, prior to a print starting, rather than doing a quick sweeping scan, it does a scan where it stops at every probe point.  It doesn't z hop or anything, just goes to a probe point, stops, and then proceeds to the next.

**Comment by l3ean5** [comment]
nvm, figured it out.  Thank you.

**Reply by JasonWThompson** [reply]
@l3ean5, so what did you do? I have the macro above defined, but I'm still z-hoppin.

**Reply by l3ean5** [reply]
I'm having a hard time remembering but it looks like I modified the start_end.cfg.  Don't know if I added or modified the following:

```
[gcode_macro _km_mesh_if_needed]
gcode:
  # TODO: Instead of blindly using the loaded mesh we could probe a few key
  # points on the saved grid. If those probes show that the saved grid is no
  # longer in tolerance we could just run BED_MESH_CALIBRATE_FAST anyway.
  {% if not printer.bed_mesh.profile_name %}
    BED_MESH_CALIBRATE METHOD=scan SCAN_MODE=rapid {rawparams}
  {% endif %}
```

comparing it to my other printer it looks like I modified line 303ish

**Comment by l3ean5** [comment, marked as answer]
I think I figured it out for sure this time:

I uncommented "[include klipper-macros/optional/bed_mesh.cfg]" in my printer.cfg and now it scans.  I'm dumb.
